x-pack/plugins/security/server/authentication/providers/pki.ts

```diff
@@ -255,23 +261,53 @@ export class PKIAuthenticationProvider extends BaseAuthenticationProvider {
    */
   private getCertificateChain(peerCertificate: DetailedPeerCertificate | null) {
     const certificateChain = [];
+    const certificateStrings = [];
+    let isChainIncomplete = false;
     let certificate: DetailedPeerCertificate | null = peerCertificate;
-    while (certificate !== null && Object.keys(certificate).length > 0) {
+
+    while (certificate && Object.keys(certificate).length > 0) {
       certificateChain.push(certificate.raw.toString('base64'));
+      certificateStrings.push(this.stringifyCertificate(certificate));
 
       // For self-signed certificates, `issuerCertificate` may be a circular reference.
       if (certificate === certificate.issuerCertificate) {
         this.logger.debug('Self-signed certificate is detected in certificate chain');
-        certificate = null;
+        break;
+      } else if (certificate.issuerCertificate === undefined) {
+        // The chain is only considered to be incomplete if one or more issuerCertificate values is undefined;
+        // this is not an expected return value from Node, but it can happen in some edge cases
+        isChainIncomplete = true;
+        break;
       } else {
+        // Repeat the loop
         certificate = certificate.issuerCertificate;
       }
     }
 
-    this.logger.debug(
-      `Peer certificate chain consists of ${certificateChain.length} certificates.`
-    );
+    this.logger.debug(`Peer certificate chain: [${certificateStrings.join(', ')}]`);
+
+    return { certificateChain, isChainIncomplete };
+  }
+
+  private stringifyCertificate(peerCertificate: DetailedPeerCertificate) {
```

Comment:
__optional nit:__ it can be an external function like `canStartNewSession` since it doesn't require `this`.